Backend/Remora.Discord.API.Abstractions/API/Objects/AutoModeration/IAutoModerationActionMetadata.cs

```diff
@@ -0,0 +1,49 @@
+//
+//  IAutoModerationActionMetadata.cs
+//
+//  Author:
+//       Jarl Gullberg <jarl.gullberg@gmail.com>
+//
+//  Copyright (c) 2017 Jarl Gullberg
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+using Remora.Rest.Core;
+
+namespace Remora.Discord.API.Abstractions.Objects.AutoModeration;
+
+/// <summary>
+/// Additional data used when an action is executed. Different fields are relevant based on value of the
+/// <see cref="AutoModerationActionType"/>.
+/// </summary>
+public interface IAutoModerationActionMetadata
+{
+    /// <summary>
+    /// Gets the channel to which user content should be logged.
+    /// </summary>
+    /// <remarks>
+    /// The associated action type for this property is <see cref="AutoModerationActionType.SendAlertMessage"/>.
+    /// </remarks>
+    Optional<Snowflake> ChannelID { get; }
```

Comment:
If these fields are indeed optional, perhaps an additional PR to the Discord documentation would be appropriate.

Reply:
I'm not sure if the folks there will agree on me with this one, but I'll give it a shot: https://github.com/discord/discord-api-docs/pull/5090

Reply:
I do actually the type of the action metadata as
```cs Optional<OneOf<...SendAlertMessageMetadata,...TimeoutMetadata>>
```

But as in your tests stated in your PR it doesn't behave like it.